core/consensus/slots/src/slots.rs

```diff
@@ -96,12 +99,13 @@ impl SlotInfo {
 }
 
 /// A stream that returns every time there is a new slot.
-pub struct Slots<SC> {
+pub(crate) struct Slots<SC> {
 	last_slot: u64,
 	slot_duration: u64,
 	inner_delay: Option<Delay>,
 	inherent_data_providers: InherentDataProviders,
 	timestamp_extractor: SC,
+	slots_per_epoch: NonZeroU64,
```

Comment:
this isn't really helpful for Aura. can it be done in BABE only?

Reply:
This is requiring that any slot-based consensus have a notion of epoch demarcated by slot

Reply:
> This is requiring that any slot-based consensus have a notion of epoch demarcated by slot

I agree.  That said, this is enough to get BABE working.  Is it okay if I get BABE complete first, since that is what the product needs?

Reply:
Sure, let's file something for later.